### DISCUSSION TITLE
How to update slides when deleting the last image from list?

### DISCUSSION THREAD

**Discussion by sapkauskasm** [igordanchenko/yet-another-react-lightbox#40] [Q&A] [answered]
Let's say I have an array of images, and want to delete images from a light box.

Take this sample:
https://codesandbox.io/s/yet-another-react-lightbox-examples-forked-6itfhp?file=/src/examples/PhotoGallery.js

It is working as expected, until you try to delete the last image of the array.
Just open the example, select the last image and try to delete it.

Is it possible, that  the length of  the slides is not always being updated?

**Comment by igordanchenko** [comment, marked as answer]
I'll be totally honest with you, the runtime modification of the slides array is not a use case I had in mind while designing the lightbox :)

But I did some testing, and it looks like you can still make it work (though the implementation isn't trivial). The idea here is to jump one slide back  before deleting the very last slide.

https://codesandbox.io/s/yet-another-react-lightbox-40-54cioq?file=/src/examples/PhotoGallery.js

--

**v3 UPDATE** 

v3 now officially supports this use case. Here is more elegant implementation for v3.

https://codesandbox.io/s/yet-another-react-lightbox-40-v3-dy1ngu?file=/src/examples/PhotoGallery.js
